src/main/java/com/meta/cp4m/message/Message.java

```diff
@@ -32,13 +36,26 @@ static Identifier threadId(Identifier id1, Identifier id2) {
 
   Role role();
 
+  @Nullable Message parentMessage();
+
   default Identifier threadId() {
     return threadId(senderId(), recipientId());
   }
 
   enum Role {
-    ASSISTANT,
-    USER,
-    SYSTEM
+    ASSISTANT(0),
+    USER(1),
+    SYSTEM(2);
+
+    private final int priority;
+
+    private Role(Integer priority){
+      this.priority = priority;
+    }
+
+    public int getPriority(){
```

Comment:
```suggestion
    public int priority(){
```

I never told you but I've been using a fluent api style so lets stick with that